packages/xstate-solid/test/useMachine-fsm.test.tsx

```diff
@@ -0,0 +1,686 @@
+
+    expect(machine1Value.textContent).toEqual('100');
+    expect(machine2Value.textContent).toEqual('100');
+
+    fireEvent.click(incMachine1);
+
+    expect(machine1Value.textContent).toEqual('101');
+    expect(machine2Value.textContent).toEqual('100');
+
+    fireEvent.click(incMachine2);
+
+    expect(machine1Value.textContent).toEqual('101');
+    expect(machine2Value.textContent).toEqual('101');
+    done();
+  });
+
+  it('send should update synchronously', (done) => {
+    const machine = createMachine({
+      initial: 'start',
+      states: {
+        start: {
+          on: {
+            done: 'success'
+          }
+        },
+        success: {}
+      }
+    });
+
+    const Spawner = () => {
+      const [current, send] = useMachine(machine);
+      onMount(() => {
+        expect(current.value).toBe('start');
+        send({ type: 'done' });
+        expect(current.value).toBe('success');
+      });
+
+      return (
+        <Switch fallback={null}>
+          <Match when={current.value === 'start'}>
+            <span data-testid="start" />
+          </Match>
+          <Match when={current.value === 'success'}>
+            <span data-testid="success" />
+          </Match>
+        </Switch>
+      );
+    };
+
+    render(() => <Spawner />);
+    waitFor(() => screen.getByTestId('success')).then(() => done());
+  });
+
+  // Example from: https://github.com/davidkpiano/xstate/discussions/1944
+  it('fsm useMachine service should be typed correctly', () => {
+    interface Context {
+      count?: number;
+    }
+
+    interface Event {
+      type: 'READY';
+    }
+
+    type State =
+      | {
+          value: 'idle';
+          context: Context & {
+            count: undefined;
+          };
+        }
+      | {
+          value: 'ready';
+          context: Context & {
+            count: number;
+          };
+        };
+
+    type Service = StateMachine.Service<Context, Event, State>;
+
+    const machine = createMachine<Context, Event, State>({
+      id: 'machine',
+      initial: 'idle',
+      context: {
+        count: undefined
+      },
+      states: {
+        idle: {},
+        ready: {}
+      }
+    });
+
+    const useCustomService = (service: Service) => createEffect(() => service);
+
+    function App() {
+      const [, , service] = useMachine(machine);
+
+      useCustomService(service);
+
+      return null;
+    }
+
+    const noop = (_val: any) => {
+      /* ... */
+    };
+
+    noop(App);
+  });
+
+  it('Service should stop on component cleanup', () => {
+    const machine = createMachine({
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            EV: {
+              target: 'b'
+            }
+          }
+        },
+        b: {}
+      }
+    });
+
+    const Display = () => {
+      const [state, , service] = useMachine(machine);
+      onCleanup(() => {
+        expect(service.status).toBe(InterpreterStatus.Stopped);
```

Comment:
I think that `done()` is missing here. When this test completes its sync execution Jest has no idea that all assertions were not yet executed

Reply:
this was resolved but the current version still suffers from this problem